### DISCUSSION TITLE
Use vueuse to rely less on bootstrap js

### DISCUSSION THREAD

**Discussion by xvaara** [cdmoro/bootstrap-vue-3#197] [Ideas]
Was just upgrading stuff to vue 3 and played (more) with vueuse. 

It has a wonderful library of functions for common need. For example modal backdrop or dropdown out of element click with one line: https://vueuse.org/core/onClickOutside/

You (we?) should look into using it more and maybe write out bootstrap dom manipulation in favour of vue functionality.

**Comment by xvaara** [comment]
I did a color picker and the functionality of the "color area" was just this:
```js
const area = ref(null)
const colorAreaDims = reactive(useElementBounding(area))

const { elementX, elementY, isOutside } = useMouseInElement(area)
const { pressed } = useMousePressed({ target: area })

watch([elementX, elementY, pressed], ([x, y, p]) => {
  if (p && !isOutside.value && colorAreaDims.width !== 0) {
    hsva.s = x / colorAreaDims.width * 100
    hsva.v = 100 - (y / colorAreaDims.height * 100)
    emit('update:modelValue', hex.value)
  }
})
```

**Comment by jclijmans** [comment]
I was looking into this as well for the debounce functionality. Seems a nice set of reusable functions

**Reply by xvaara** [reply]
and most of the devs are vue (and nuxt) core team members so you can hope they are well made 🤣
